tests/poll.rs

```diff
@@ -201,10 +208,70 @@ fn registry_behind_arc() {
     handle2.join().unwrap();
 }
 
-// On kqueue platforms registering twice (not *re*registering) works.
+// This test checks the following reregister constraints:
+// The `reregister` arguments fully override the previous values. In other
+// words, if a socket is registered with [`readable`] interest and the call
+// to `reregister` specifies [`writable`], then read interest is no longer
+// requested for the handle.
+#[test]
+fn reregister_overrides_interest() {
+    let (mut poll, mut events) = init_with_poll();
+
+    let barrier = Arc::new(Barrier::new(2));
+    let (thread_handle, address) = start_listener(1, Some(barrier.clone()), false);
+
+    let mut stream = TcpStream::connect(address).unwrap();
+
+    poll.registry()
+        .register(&mut stream, ID1, Interest::WRITABLE)
+        .expect("unable to register TCP stream");
+
+    poll.registry()
+        .reregister(&mut stream, ID2, Interest::READABLE)
+        .expect("unable to register TCP stream");
+
+    expect_no_events(&mut poll, &mut events);
+
+    barrier.wait();
+    thread_handle.join().expect("unable to join thread");
+}
+
+// This test checks the following constraints:
+// - reregister can use the same token as register
+// - reregister can use different token from register
+// - multiple reregister are ok
+#[test]
+fn register_reregister_token_usage() {
+    let poll = Poll::new().expect("unable to create Poll instance");
+
+    let mut listener1 = TcpListener::bind(any_local_address()).unwrap();
+
+    poll.registry()
+        .register(&mut listener1, ID1, Interest::READABLE)
+        .expect("unable to register listener");
+
+    poll.registry()
+        .reregister(&mut listener1, ID1, Interest::READABLE)
+        .expect("unable to register listener");
+
+    poll.registry()
+        .reregister(&mut listener1, ID2, Interest::READABLE)
+        .expect("unable to register listener");
+}
+
+// This test checks the following register constraint:
+// The event source must **not** have been previously registered with this
+// instance of `Poll`, otherwise the behavior is undefined.
+//
+// This test is done on Windows only because Windows is the only platform where
```

Comment:
In which way you think `Linux` should be mentioned? If we consider mentioning `Linux` then we should mention it for all 3 cases: double register(this one), reregister w/o register and deregister w/o register. The behavior seem to be the same for all 3 of them. Thoughts?

Reply:
Similar to the above, I think for a comment being *added* we can just make the change here.

Something like: "This test is done only on Windows and epoll platforms because registering a source twice is defined behavior that fails with an error".

Reply:
ok, got it